### DISCUSSION TITLE
Incremental Local Outlier Factor (ILOF)

### DISCUSSION THREAD

**Discussion by pietro-tanure** [online-ml/river#1231] [General]
I've implemented a class for Incremental Local Outlier Factor since river didn't have this, based on the algorithm defined in the article: 
Incremental Local Outlier Detection for Data Streams
[Dragoljub Pokrajac]
https://ieeexplore.ieee.org/document/4221341

I think it can offer some good foundations to further continuation of incremental LOF algorithms like DILOF, MILOF, GILOF and etc. 

I didn't use the Nearest Neighbor class already defined by river as it didn't store the distance to its neighbors for each new point, nor the reverse neighbors and other relevant information for the ILOF algorithm, which would need to be constantly recalculated making the algorithm ineficient.

**Comment by MaxHalford** [comment]
Good stuff! I didn't have time to look in depth, I just skimmed through. My first feedback is that it seems the algorithm is a mini-batch algorithm. What you've done is that you store the samples until enough samples have been seen. That's not really the idiomatic way to proceed. The better approach would be to implement a `learn_many` function, which takes as input a `pd.DataFrame`. See what I mean?

**Reply by pietro-tanure** [reply]
I updated the code with the learn_many using pandas as you said
